docs/agents/spec-process.md

```diff
@@ -0,0 +1,58 @@
+  just like in code reviews.
+- If there are changes or amendments to a spec later on,
+  a PR to an existing spec is much cleaner than having to track down all preceding issues to find out what the current state is.
+- Easier to get a new agent (in-house or community) started.
+
+## Process
+
+1. Agents discussion issue \
+  If there's a lot of uncertainty about what the poposal is,
+  a discussion issue should be created.
+  In doubt, lean towards not creating a discussion issue and start with a draft PR.
+  Discussion issues should not have votes/agent checkboxes.
+  It's not expected that all agents participate in the initial discussions, although everyone is invited to do so.
+1. Draft PR \
+  It's fine if the first draft of the spec change even if details are unclear and need discussion.
+  The PR should include which agents are affected by this change.
+  Specifically, it should always be clear whether the RUM agent is affected by this.
+  This doesn't have to be determined yet when creating the draft PR,
+  but it is required before marking the PR as `Ready for Review`.
+1. Approval by at least one other agent \
+  This makes sure the change makes sense for not just one agent.
+  This doesn't ping all agent teams yet,
+  which only happens in the next step.
+  Of course,
+  every agent is more than welcome to contribute at any point in this process,
+  but they are not expected to actively keep track if it.
+1. Ready for Review \
+  This automatically requests review from all agents due to the [`CODEOWNERS`](https://github.com/elastic/apm/tree/master/.github/CODEOWNERS) file.
+  After a quorum of agents has approved, others have 1 week to veto by requesting changes.
+  If the proposed changes are likely to be problematic for a certain agent,
+  an approval from that agent is mandatory.
+1. Prioritization \
+  At any point in time,
+  but before merging the PR,
+  the priority should be determined.
+  That happens in collaboration with devs and leads.
+  It's recommended to do that sooner rather than later
+  so that we don't spend a lot of time discussing a feature that's not going to be prioritized.
+  As a result of the prioritization,
+  the PR gets assigned to a milestone,
+  reflecting the stack release version this feature should be implemented by all agents.
+  If some agents are not going to implement the change by that stack release version,
+  or if there are no immediate plans to implement the change at all,
+  a comment should be made on the PR.
+1. PR can be merged by the author of the PR \
+  Three business days before merging, there should be a last call for objections.
+1. The author of the PR creates an issue in each agent's repo \
+  They make sure the issues are assigned to a stack release milestone,
+  based on the milestone of the spec PR.
+  Agent teams can still change the milestone if priorities shift.
+  However, they should talk to their team lead about it.
```

Comment:
```suggestion
  However, this should be discussed in the specific agent's planning meeting and mentioned in the agent (bi-)weekly meeting.
```

I think instead of chasing a team lead, it's better to have a discussion in the agent meeting, we sometimes also have PMs there and if we mention it in the agents weekly, then the person who opened the PR will be also informed about it. Of course team lead can jump in at any time.

Reply:
The issue I see is that currently, the agent meetings are bi-weekly. So there could be 2 weeks before discussing an issue might slip into another release. Also, it's likely not too relevant for agent team A that agent team B implements a change a bit later.

Reply:
I totally agree on the bi-weekly part and the low relevance to other teams (that's the 2. part of the suggestion). My main point (the 1. part of the suggested change) was that I prefer these to be team decisions and not be by default a single person decision because in my experience that introduces bottleneck and does not improve the quality of the decision. 

Also the specific agent's planning meeting is in my opinion a really good setup for these decisions, we have products + dev + maybe a team lead there. I feel a team lead can always jump in at any point and force a decision (they don't need special permissions in the process, they have it anyways :) ); so rather than leaving this by default to a team lead, I'd try to encourage the specific agent team with all other stakeholders (eg. product) to make a reasonable decision in the agent's planning meeting.

Reply:
> I prefer these to be team decisions and not be by default a single person decision because in my experience that introduces bottleneck and does not improve the quality of the decision.

Totally agree. It should mainly be the team's decision. They should just let their team lead know which could be in the form of a slack message or in a weekly meeting. I wouldn't want to be too specific to allow for flexibility.

What do you think of changing the wording to 
> However, they should let their team lead know.

The thought behind it is that the team lead can then inform PMs if necessary so that devs don't have to chase them. I wouldn't want to assume that there's always a PM on the agent weeklies.

Reply:
> What do you think of changing the wording to 
>> However, they should let their team lead know.

That sounds very good to me. 👍